File: Boot/Brasserie/src/main/java/soprajc/Brasserie/model/Note.java
package soprajc.Brasserie.model;

import java.util.Objects;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.ForeignKey;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.OneToOne;
import javax.persistence.Table;
import javax.persistence.Version;
import javax.validation.constraints.NotNull;

import com.fasterxml.jackson.annotation.JsonView;

@Entity
@Table(name="note")
public class Note {
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@JsonView(JsonViews.Common.class)
	private Integer id_note;
	
	@OneToOne
	@JoinColumn(name="id_client", foreignKey = @ForeignKey(name="note_client_fk"), nullable = true)
	@JsonView(JsonViews.Note.class)
	private Client client;
	
	@ManyToOne
	@JoinColumn(name= "id_biere", foreignKey = @ForeignKey(name="note_biere_fk"))
	@JsonView(JsonViews.Note.class)
	@NotNull
	private Biere biere;
	
	@JsonView(JsonViews.Common.class)
	@NotNull
	private Double note;
	
	@JsonView(JsonViews.Common.class)
	@Column(length=300)
	private String commentaire;
	
	@Version
	protected int version;
	
	public Note() {}
	
	public Note(Client client, Biere biere, Double note) 
	{
		this.client = client;
		this.biere= biere;
		this.note=note;
	}
	
	public Note(Client client, Biere biere, Double note, String commentaire) 
	{
		this.client = client;
		this.biere= biere;
		this.note=note;
		this.commentaire = commentaire;
	}

	public Integer getId_note() {
		return id_note;
	}

	public void setId_note(Integer id_note) {
		this.id_note = id_note;
	}
	
	public Client getClient() {
		return client;
	}

	public void setClient(Client client) {
		this.client = client;
	}

	public Biere getBiere() {
		return biere;
	}

	public void setBiere(Biere biere) {
		this.biere = biere;
	}

	public Double getNote() {
		return note;
	}

	public void setNote(Double note) {
		this.note = note;
	}
	
	public String getCommentaire() {
		return commentaire;
	}

	public void setCommentaire(String commentaire) {
		this.commentaire = commentaire;
	}

	public int getVersion() {
		return version;
	}

	public void setVersion(int version) {
		this.version = version;
	}

	@Override
	public int hashCode() {
		return Objects.hash(id_note);
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		Note other = (Note) obj;
		return Objects.equals(id_note, other.id_note);
	}
}

File: Angular/Brasserie/src/app/model/note.ts
import { Biere } from './biere';
import { Client } from './client';
export class Note {
  public constructor(
    private _id_note?: number | undefined,
    private _id_client?: Client | undefined,
    private _biere?: Biere | undefined,
    private _note?: number | undefined,
    private _commentaire?: string | undefined
  ) {}

  public get id_note(): number | undefined {
    return this._id_note;
  }

  public set id_note(value: number | undefined) {
    this._id_note = value;
  }

  public get id_client(): Client | undefined {
    return this._id_client;
  }

  public set id_client(value: Client | undefined) {
    this._id_client = value;
  }

  public get biere(): Biere | undefined {
    return this._biere;
  }

  public set biere(value: Biere | undefined) {
    this._biere = value;
  }

  public get note(): number | undefined {
    return this._note;
  }

  public set note(value: number | undefined) {
    this._note = value;
  }

  public get commentaire(): string | undefined {
    return this._commentaire;
  }

  public set commentaire(value: string | undefined) {
    this._commentaire = value;
  }
}
